src/main/java/edu/escuelaing/ieti/matchmaking/persistence/UserRepository.java

```diff
@@ -0,0 +1,19 @@
+package edu.escuelaing.ieti.matchmaking.persistence;
+
+import edu.escuelaing.ieti.matchmaking.model.User;
+
+import java.util.List;
+
+public interface UserRepository {
```

Comment:
Consider using the  CrudRepository interface instead of your own interface.  That will provide a lot of functionality out-of-the box.